### DISCUSSION TITLE
Hanging builds in matrix?

### DISCUSSION THREAD

**Discussion by jGaboardi** [community/community#25348] [Actions]
<p>I am curious about why some builds within a job matrix hang. These jobs run on a matrix of operating systems, Python versions, and environments. They seem to hang nearly randomly without regard to OS, but more often with Python 3.8 and Windows. Although I have only been working with GitHub Actions for a few days, this seems to happen about half the time with 1 or 2 builds in the matrix each time the workflow job is run.</p>
<p>I am wondering if there is something I am doing incorrectly or something that I can adjust in order for the builds to run smoothly? </p>
<ul>
<li>
<strong>repo</strong> — <a href="https://github.com/pysal/spaghetti" rel="nofollow noopener">pysal/spaghetti</a>
</li>
<li>
<strong>workflow</strong>
<ul>
<li><a href="https://github.com/pysal/spaghetti/blob/master/.github/workflows/spaghetti-unittests.yml" rel="nofollow noopener">master version</a></li>
<li><a href="https://github.com/pysal/spaghetti/pull/453/files#diff-6dacb6da9e0e22d2c423ff68bcd6744e" rel="nofollow noopener">current PR verison</a></li>
</ul>
</li>
<li>
<strong>build log</strong>
<ul>
<li><a href="https://github.com/pysal/spaghetti/actions/runs/63093178" rel="nofollow noopener">master version</a></li>
<li><a href="https://github.com/pysal/spaghetti/actions/runs/63179512" rel="nofollow noopener">current PR version</a></li>
</ul>
</li>
</ul>

**Comment by Yanjingzhu** [comment]
<p>I checked your workflow runs and forked your repo to my organization, I got the same result, some of the matrix jobs with python 3.8 hungs randomly. I have reported your issue to engineering team, I will keep you posted as quickly as possible. </p>

**Comment by jGaboardi** [comment]
<p>Thanks! I’m looking forward to hearing back from you.</p>

**Comment by jGaboardi** [comment]
<p><strong>Update</strong></p>
<p>The <a href="https://github.com/pysal/spaghetti/runs/541926057" rel="nofollow noopener">spaghetti workflows</a> are still failing randomly, but I have implemented a similar workflow in another of our repos (<a href="https://github.com/pysal/libpysal" rel="nofollow noopener">libpysal</a>), which runs successfully without stalling out. In it we are testing Python {3.6,7,8} and Mac, Unbuntu, and Windows, though we are testing with fewer environments than in <a href="https://github.com/pysal/spaghetti/blob/master/.github/workflows/unittests.yml" rel="nofollow noopener">spaghetti</a>.</p>
<p>libpysal’s <a href="https://github.com/pysal/libpysal/blob/master/.github/workflows/unittests.yml" rel="nofollow noopener">unittests.yml</a></p>
<p>libpysal’s <a href="https://github.com/pysal/libpysal/tree/master/ci" rel="nofollow noopener">environments</a></p>

**Comment by jGaboardi** [comment]
<p><strong>Update</strong></p>
<p>libpysal builds now failing. See <a href="https://github.community/t/error-in-python-testing-environment/17984">here</a>.</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/jgaboardi">@jgaboardi</a> Thank you for your updates. The engineering team is still investigating this issue,  they still need some time to find the cause. Your patience and understanding will be appreciated. </p>

**Comment by jGaboardi** [comment]
<p>No problem, and thanks!</p>

**Comment by jGaboardi** [comment, marked as answer]
<p>This seems to have worked itself out now, though no clear solution was ever presented.</p>
